specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx

```diff
@@ -63,14 +64,20 @@ export function PrepDialog({
     return mutatedPreparations as RA<PreparationData>;
   }, [rawPreparations, itemCollection]);
 
-  const [selected, setSelected] = useLiveState<RA<number>>(
-    React.useCallback(
-      () => Array.from({ length: preparations.length }).fill(0),
-      [preparations.length]
-    )
+  // Change to use an object for selected state allowing null values
+  const [selected, setSelected] = useLiveState<{
+    [key: string]: number | null;
```

Comment:
(optional) instead of null, could you refactor the code you added in this file to use `undefined`? that would be more consistent with what we use in the rest of the codebase